### PR TITLE
Rdcc 5572 update case flags api

### PR DESCRIPTION
**Pull request by vilasshelke-hmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-5572

### Change description ###

Case Flags : Update the Get Flags API 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by nayeemshaik-hmcts — APPROVED** [review]
changes looks fine

File: src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCaseFlagsIntegrationTest.java [hunk 1]
```diff
@@ -115,13 +116,133 @@ void shouldRetrieveCaseFlagForWelshRequiredasWithStatusCode_404()
...
+            path
+        );
+        assertEquals(2, response.getFlags().get(0).getFlagDetails().size());
+        List<FlagDetail> flagDetailList = response.getFlags().get(0).getFlagDetails();
+        for (FlagDetail flagDetail : flagDetailList) {
+            if (flagDetail.getParent()) {
+                for (FlagDetail childFlag : flagDetail.getChildFlags()) {
+                    //child flag
+                    if (childFlag.getFlagCode().equalsIgnoreCase("PF0027")) {
+                        assertEquals("Test27", childFlag.getNameCy());
```
Comment (rajkatla-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCaseFlagsIntegrationTest.java:185]: add assertions for other fields as well

File: src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCaseFlagsIntegrationTest.java [hunk 2]
```diff
@@ -115,13 +116,133 @@ void shouldRetrieveCaseFlagForWelshRequiredasWithStatusCode_404()
...
+
+    @Test
+    void shouldReturnSuccessForRetrieveCaseFlagsByServiceIdWithAvailableExternalFlagIsN()
+        throws JsonProcessingException {
+        final var response = (CaseFlag) commonDataApiClient.retrieveCaseFlagsByServiceId(
+            "AAA1?available-external-flag=N",
+            CaseFlag.class,
+            path
+        );
+        assertEquals(2, response.getFlags().get(0).getFlagDetails().size());
```
Comment (rajkatla-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCaseFlagsIntegrationTest.java:225]: add assertions for response fields

File: src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCaseFlagsIntegrationTest.java [hunk 3]
```diff
@@ -115,13 +116,133 @@ void shouldRetrieveCaseFlagForWelshRequiredasWithStatusCode_404()
...
+
+    @Test
+    void shouldReturnSuccessForRetrieveCaseFlagsByServiceIdWithAvailableExternalFlagIsY()
+        throws JsonProcessingException {
+        final var response = (CaseFlag) commonDataApiClient.retrieveCaseFlagsByServiceId(
+            "AAA1?flag-type=PARTY&available-external-flag=Y",
+            CaseFlag.class,
+            path
+        );
+        assertEquals(1, response.getFlags().get(0).getFlagDetails().size());
```
Comment (rajkatla-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCaseFlagsIntegrationTest.java:236]: add assertions for response fields

File: src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCaseFlagsIntegrationTest.java [hunk 4]
```diff
@@ -115,13 +116,133 @@ void shouldRetrieveCaseFlagForWelshRequiredasWithStatusCode_404()
...
+        final var response = (CaseFlag) commonDataApiClient.retrieveCaseFlagsByServiceId(
+            "AAA1?welsh-required=Y",
+            CaseFlag.class,
+            path
+        );
+        assertEquals(2, response.getFlags().get(0).getFlagDetails().size());
+    }
+
+    @Test
+    void shouldReturnSuccessForRetrieveCaseFlagsByServiceIdWithWelshFlagIsN() throws JsonProcessingException {
```
Comment (rajkatla-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCaseFlagsIntegrationTest.java:208]: this test case already covered in line no 198 test method

File: src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCaseFlagsIntegrationTest.java [hunk 5]
```diff
@@ -115,13 +116,133 @@ void shouldRetrieveCaseFlagForWelshRequiredasWithStatusCode_404()
...
+    void shouldReturnSuccessForRetrieveCaseFlagsByServiceIdWithAvailableExternalFlagIsY()
+        throws JsonProcessingException {
+        final var response = (CaseFlag) commonDataApiClient.retrieveCaseFlagsByServiceId(
+            "AAA1?flag-type=PARTY&available-external-flag=Y",
+            CaseFlag.class,
+            path
+        );
+        assertEquals(1, response.getFlags().get(0).getFlagDetails().size());
+    }
+
```
Comment (rajkatla-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCaseFlagsIntegrationTest.java:238]: write another scenario to cover both available-external-flag=Y  and welsh-required=Y

File: src/functionalTest/java/uk/gov/hmcts/reform/cdapi/CommonDataApiFunctionalTest.java [hunk 6]
```diff
@@ -344,4 +344,39 @@ void shouldReturnSuccess_Empty_ServiceID() {
...
+    }
+
+    @Test
+    @ToggleEnable(mapKey = MAP_KEY_CASE_FLAGS, withFeature = true)
+    @ExtendWith(FeatureToggleConditionExtension.class)
+    void shouldReturnNotFoundForRetrieveCaseFlagsByServiceIdWithAvailableExternalFlagAndDefaultStatus() {
+        final var response = (ErrorResponse)
+            commonDataApiClient.retrieveCaseFlagsByServiceId(
+                NOT_FOUND,
+                "service-id=XBBB?available-external-flag=Y&default-status=true"
```
Comment (rajkatla-hmcts) [inline src/functionalTest/java/uk/gov/hmcts/reform/cdapi/CommonDataApiFunctionalTest.java:377]: endpoint not expecting default-status as query param , what's use of this test case ?

**Review by rajkatla-hmcts — CHANGES_REQUESTED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCaseFlagsIntegrationTest.java [hunk 7]
```diff
@@ -115,13 +116,133 @@ void shouldRetrieveCaseFlagForWelshRequiredasWithStatusCode_404()
...
+            CaseFlag.class,
+            path
+        );
+        assertEquals(2, response.getFlags().get(0).getFlagDetails().size());
+        verifyResponse(response.getFlags().get(0).getFlagDetails());
+        List<FlagDetail> flagDetailList = response.getFlags().get(0).getFlagDetails();
+        for (FlagDetail flagDetail : flagDetailList) {
+            assertEquals(false, flagDetail.getExternallyAvailable() == null);
+            assertEquals(false, flagDetail.getDefaultStatus() == null);
+            if (flagDetail.getParent()) {
```
[the remark this answers is not in the document]
Comment (vilasshelke-hmcts) [reply src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCaseFlagsIntegrationTest.java:134]: Accommodated the changes

File: src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCaseFlagsIntegrationTest.java [hunk 8]
```diff
@@ -115,13 +116,133 @@ void shouldRetrieveCaseFlagForWelshRequiredasWithStatusCode_404()
...
+                flagDetail.getChildFlags().forEach(cf -> assertNull(cf.getNameCy()));
```
[the remark this answers is not in the document]
Comment (vilasshelke-hmcts) [reply src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCaseFlagsIntegrationTest.java:135]: @rajkatla-hmcts - This is JSON response only

**Pull request closed without merging**